### ISSUE TITLE
[multistage] Projection Not Pushed Down for Agg Queries with No Grouping Sets

### ISSUE THREAD

**Issue by ankitsultana** [apache/pinot#10129]
Repro:

```
SELECT
  COUNT(*) -- the issue happens even if you have COUNT(A.playerID)
FROM
  baseballStats_OFFLINE AS A
  JOIN baseballStats_OFFLINE AS B ON A.playerID = B.playerID
WHERE
  A.hits > 10 AND B.hits < 5
```

The issue happens even if you use a sub-query:
```
SELECT
  COUNT(*)
FROM
  (
    SELECT
      A.playerID
    FROM
      baseballStats_OFFLINE AS A
      JOIN baseballStats_OFFLINE AS B ON A.playerID = B.playerID
    WHERE
      A.hits > 10
      AND B.hits < 5
  )
```

For each of the queries above, they will end up reading all the columns in the table-scan stage. The reason is that there's no projection node created by Calcite.

I am testing out a few approaches for a fix in this PR: https://github.com/apache/pinot/pull/10122

Btw, this is what GPT recommends:

cc: @walterddr 

<img width="786" alt="image" src="https://user-images.githubusercontent.com/8644710/212462065-5cec3444-31d4-4965-b4f3-f771303a1fa5.png">

**Comment by walterddr** [comment]
sounds like a good idea. :-P

**Comment by walterddr** [comment]
we can leverage calcite's AggregateExtractProjectRule and split the aggregate with aggregate + project. 

this only needed for count(*), i realized that sum and others are properly putting the project in place

**Comment by walterddr** [comment]
did a bit of research on this topic:

the problem only occurs on `COUNT(*)` or `COUNT(1)`, `COUNT(col)` this is b/c calcite will optimize out the argument since all columns in pinot are marked non-nullable. 

1. several attempts have been tried on this one but the most promising so far is to replace `COUNT` with `SUM(1)` thus bypass calcite's optimization. with this project pushdown are being executed properly.
2. @ankitsultana also reported that the pushdown doesn't apply to semi JOINs with NOT-IN clause properly
 ```
SELECT
  SUM(1)
FROM
  baseballStats_OFFLINE
WHERE
  hits > 10 AND 
  playerID NOT IN (SELECT playerID FROM baseballStats_OFFLINE WHERE hits < 5)
```
^ this seems to be related to the fact that NOT IN produces a in-equality JOIN condition that cannot be optimized into a proper JOIN and thus cannot apply proper pushdown rules. 
which we will address separately.

**Comment by ankitsultana** [comment]
@walterddr : Can we consider using `RelFieldTrimmer`?

I raised this: https://github.com/apache/pinot/pull/10156

It seems to work and I am testing it out on our clusters. The PR is not final as I am also reading more about this feature. There are some caveats called out in the code also regarding this.

**Comment by walterddr** [comment]
is this issue fixed by #10187 ?

**Comment by ankitsultana** [comment]
Yes I think we can close this now.
